#
# This is a Shiny web application. You can run the application by clicking
# the 'Run App' button above.
#
# Find out more about building applications with Shiny here:
#
#    http://shiny.rstudio.com/
#

library(shiny)
library(vcfR)

# Define UI for application that draws a histogram
ui <- fluidPage(

    # Application title
    titlePanel("BIOS611 Ashbya genomes project"),

    sidebarLayout(
        sidebarPanel(
            fileInput("vcf", "Provide a variant call (vcf) File", accept = ".vcf"),
            fileInput("fasta", "And a genome (fasta) File", accept = ".fa"),
            fileInput("gff", "And an annotation (gff) File", accept = ".gff3"),
            actionButton("make_plot", "Plot")
        ),
    mainPanel(
        plotOutput("contents")
        )
    )

)

# Define server logic required to draw a histogram
server <- function(input, output, session) {
    
    # By default shiny does not allow files >5MB. 
    options(shiny.maxRequestSize=30*1024^2)
    data_input <- reactive({
        req(input$csv_input)
        fread(input$csv_input$datapath)
    })
    
    
    vcf <- reactive({req(input$vcf)
        read.vcfR(input$vcf)})
    #gen <- vcfR2genlight(vcf)
    dna <- reactive({req(input$dna)
        ape::read.dna(input$fasta, format = "fasta")})
    gff <- reactive({req(input$gff)
        read.table(input$gff, sep="\t", quote="")})
    plot1 <- observeEvent(input$make_plot,{
        chrom <- create.chromR(name="Supercontig", vcf=vcf(), seq=dna(), ann=gff(), verbose=TRUE)
        print("debugging")
        plot(chrom)})
    output$contents <- renderPlot(plot1)
}


# Run the application 
shinyApp(ui = ui, server = server)
